horovod/ray/elastic.py

```diff
@@ -271,14 +370,17 @@ def worker_loop(slot_info, events):
                         ray.kill(worker)
                         result = 1, time.time()
                 except Exception as e:
-                    logger.exception(str(e))
-                    # Fail
+                    logger.error(f"{slot_info.hostname}[{slot_info.rank}]:{e}")
+                    ray.kill(worker)
                     result = 1, time.time()
+            logger.debug(f"Worker ({slot_info}) routine is done!")
             return result
 
         return worker_loop
 
-    def run(self, worker_fn: Callable) -> List[Any]:
+    def run(self,
+            worker_fn: Callable,
+            callbacks: Optional[List[Callable]] = None) -> List[Any]:
```

Comment:
Can you add a docstring comment for the callbacks?

Reply:
done!